tests/virt_autotest/libvirt_extend_storage_lvm.pm

```diff
@@ -0,0 +1,143 @@
+    validate_script_output("vgdisplay lvm_vg", sub { m/lvm_vg/ }, $timeout);
+
+    ## About LVM volume group storage pool management
+    # Use an LVM Volume Group (VG) as a storage pool named 'guest_image_lvm'
+    record_info "LVM Storage Pool define";
+    assert_script_run "virsh pool-define-as guest_image_lvm logical --source-name lvm_vg --target /dev/lvm_vg";
+
+    record_info "LVM Storage Pool list";
+    assert_script_run "virsh pool-list --all | grep guest_image_lvm";
+
+    record_info "LVM Storage Pool start";
+    assert_script_run "virsh pool-start guest_image_lvm";
+
+    record_info "LVM Storage Pool autostart";
+    assert_script_run "virsh pool-autostart guest_image_lvm";
+
+    record_info "LVM Storage Pool info";
+    assert_script_run "virsh pool-info guest_image_lvm";
+
+    # Create a Storage Volume from LVM storage pool
+    record_info "Create a LVM Storage Volume";
+    assert_script_run("virsh vol-create-as guest_image_lvm $_-storage 1G", 120) foreach (keys %virt_autotest::common::guests);
+
+    record_info "List LVM Storage Volumes";
+    assert_script_run("ls /dev/lvm_vg", 120) foreach (keys %virt_autotest::common::guests);
+    assert_script_run("virsh vol-list guest_image_lvm | grep $_-storage", 120) foreach (keys %virt_autotest::common::guests);
+
+    record_info "Dump LVM Storage Volumes in XML";
+    assert_script_run("virsh vol-dumpxml --pool guest_image_lvm $_-storage", 120) foreach (keys %virt_autotest::common::guests);
+
+    # Attach a Storage Volume to guest system
+    record_info "Attached";
+    my $target = (is_xen_host) ? "xvdx" : "vdx";
+    assert_script_run("virsh attach-disk --domain $_ --source `virsh vol-path --pool guest_image_lvm $_-storage` --target $target", 120) foreach (keys %virt_autotest::common::guests);
+
+    record_info "Clone";
+    assert_script_run("virsh vol-clone --pool guest_image_lvm $_-storage $_-clone", 120) foreach (keys %virt_autotest::common::guests);
+    assert_script_run("virsh vol-info --pool guest_image_lvm $_-clone", 120) foreach (keys %virt_autotest::common::guests);
+
+    ## Cleanup
+    # Detach a Storage Volume from guest system
+    record_info "Detached";
+    assert_script_run("virsh detach-disk $_ $target", 120) foreach (keys %virt_autotest::common::guests);
+
+    # Delete and Remove a Storage Volume from LVM-based storage pool
+    record_info "Remove";
+    assert_script_run("virsh vol-delete --pool guest_image_lvm $_-clone", 120) foreach (keys %virt_autotest::common::guests);
+    assert_script_run("virsh vol-delete --pool guest_image_lvm $_-storage", 120) foreach (keys %virt_autotest::common::guests);
+
+    # Destroy the LVM volume group storage pool
+    record_info "LVM Storage Pool destroy";
+    assert_script_run "virsh pool-destroy guest_image_lvm";
+    assert_script_run "virsh pool-delete guest_image_lvm";
+    assert_script_run "virsh pool-undefine guest_image_lvm";
+    assert_script_run "virsh pool-list --all";
```

Comment:
Regarding the block from line 68 to line 120, which are about storage pool management and storage hotplug, it is indeed strightforward and clear to write workflows like this. But we have similar test code in existing storage test, right? Why do not we consider extending then reusing that code?

Reply:
Good point! have the same idea. Try to do the same code change as our osd test enhancement for our storage test. 
yes, I am working on this as your mentioned.

Reply:
Do some code enhancement to extend then reuse that code both storage pool management and storage hot plugin together.

Reply:
Good job. Much higher code reuse now. 
If you'd like to pursue even higher reuse, the storage.pm and libvirt_extend_storage_lvm.pm main test procedures are similar too, which can be abstracted and reused, eg by creating new base class for them and defining the procedures in base. Doing this will be helpful to avoid code duplication when later you implement the other two new kinds of storage(iscsi and multipath). Would you like to do so? I am fine if not this time.

Reply:
> Good job. Much higher code reuse now. If you'd like to pursue even higher reuse, the storage.pm and libvirt_extend_storage_lvm.pm main test procedures are similar too, which can be abstracted and reused, eg by creating new base class for them and defining the procedures in base. Doing this will be helpful to avoid code duplication when later you implement the other two new kinds of storage(iscsi and multipath). Would you like to do so? I am fine if not this time.

@guoxuguang  Hi Leon, I hope this message reaches you. How do you decide to go?

Reply:
yes, I had thought this as your mentioned. currently, fuc virt_storage_pool_management() and virt_storage_vol_management() support both Dir and LVM Vir storage pool type.  plan to create a basic class for dir and lvm rstorage pool as the enhancement in the future. also, regarding to the other two new kinds of storage(iscsi and multipath), there are some different part for existed storage pool(e.g dir and lvm) and the other two new kinds of storage(iscsi and multipath). do not plan merge them to the existed storage.pm and libvirt_extend_storage_lvm.pm main test. still need more investigation for the other two new kinds of storage(iscsi and multipath) to implement test automation.

Reply:
@alice-suse short update,  create a new class named as `virt_storage_management()` - Basic Virtualization Storage Management, was included Basic Virtualization Storage Pool Management and Basic Virtualization Storage Volume Management together, which can be abstracted and reused.